File: amazon.py
from selenium import webdriver
import time
# driver = webdriver.Chrome()
options = webdriver.ChromeOptions()
options.add_experimental_option('excludeSwitches', ['enable-logging'])
# options.add_argument('--headless')
driver = webdriver.Chrome(options=options)

driver.get("https://www.amazon.in/")
gmail="gokuldilip.official@gmail.com"
password="#1989Amazon"

driver.find_element_by_xpath('//*[@id="nav-signin-tooltip"]/a/span').click()
driver.find_element_by_xpath('//*[@id="ap_email"]').send_keys(gmail)
driver.find_element_by_xpath('//*[@id="continue"]').click()

driver.find_element_by_xpath('//*[@id="ap_password"]').send_keys(password)
driver.find_element_by_xpath('//*[@id="signInSubmit"]').click()
time.sleep(3)

# print(driver.page_source)
with open('html.txt', 'w',encoding="utf-8") as f:
    f.write(driver.page_source)
driver.find_element_by_xpath('//*[@id="nav-orders"]/span[1]').click() #click orders

#to select the time-period
driver.find_element_by_xpath('//*[@id="a-autoid-1-announce"]').click()
driver.find_element_by_xpath('//*[@id="orderFilter_3"]').click()

l=[]
i=2
while(1):
    try:
        driver.find_element_by_xpath('//*[@id="ordersContainer"]/div[{0}]'.format(i))
        q=driver.find_element_by_xpath('//*[@id="ordersContainer"]/div[{0}]'.format(i))
        l.append(q.text.split('\n'))
        i+=1
    except:
        break
# print(l)

with open('amazon_order_details.txt', 'w') as f:
    for i in range(len(l)-1):
        f.write(str(l[i])+'\n')
driver.find_element_by_xpath('//*[@id="nav-link-accountList"]').click()
driver.close()

File: Puppeteer/Amazon Scraping/Amazon.js
const puppeteer = require('puppeteer')
const fs = require('fs/promises')  
const prompt = require("prompt-sync")({ sigint: true });
let count = 0

async function scrap(){
    // const browser = await puppeteer.launch({headless: false});
    const browser = await puppeteer.launch({ 
        headless: false,
        defaultViewport: null,
        args: ['--start-maximized'] 
  });
    const page = await browser.newPage();
    // await page.setViewport({ width: 1600, height: 1000 })       //Resizing the windows
    await page.goto("https://www.amazon.in");                 //browse
    
    let email = "gokuldilip.official@gmail.com";
    let password = "#1989Amazon";

    // let email = prompt('Enter Required email ');
    // let password = prompt('Enter Required Password ');


    await page.waitForSelector('#nav-signin-tooltip > a > span');        //wait till selector loads
    await page.click('#nav-signin-tooltip > a > span')
    // await page.waitForXPath()
    await page.waitForSelector('#ap_email');        //wait till selector loads
    await page.type('#ap_email',email);             //select and enter the email
    await page.click('#continue');

    retrieve(page,0)
    console.log("Enter email page retrieved")
    
    await page.waitForSelector('#ap_password');     //wait till selector loads
    await page.type("#ap_password",password);       // select and enter the password    
    await page.click("#signInSubmit");              

    retrieve(page,1);
    console.log("Enter page page retrieved");

    // await page.waitFor(3000);
    await page.waitForSelector('#nav-orders');
    await page.click("#nav-orders");

    retrieve(page,2);
    console.log("Orders page retrieved");

    await page.waitForSelector('#a-autoid-1-announce');
    await page.click('#a-autoid-1-announce');

    retrieve(page,3);
    console.log("Filter option src retrieved");

    await page.click('#orderFilter_3');                 //select the sub-option "2021"
    
    retrieve(page,4);
    console.log("Sub-option '2021' src retrieved");

    

    // signout!
    await page.waitForNavigation();
    await browser.close()
    // await page.waitForSelector('#nav-link-accountList');
    // await page.hover('#nav-link-accountList');
    // await page.waitForNavigation();
    // await page.waitForSelector('#nav-item-signout > span');
    // await page.click('#nav-item-signout > span');
    console.log("Website Closed")
    
}
async function retrieve(page,num){
    await page.waitForNavigation();
    let html = await page.evaluate(() => document.body.innerHTML);
    await fs.writeFile(`dumps/dump${count}.html`,html)
    count++
}
scrap()
